workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/Readers/NonBlittableReaderTests.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using Generated.Improbable.Gdk.Tests.BlittableTypes;
+using Generated.Improbable.Gdk.Tests.NonblittableTypes;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.EditmodeTests;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using Improbable.Gdk.Core.MonoBehaviours;
+using JetBrains.Annotations;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.EditmodeTests.MonoBehaviours.Readers
+{
+    [TestFixture]
+    public class NonBlittableReaderTests
+    {
+        private class SpatialOSNonBlittableComponentReader :
+            NonBlittableReaderBase<SpatialOSNonBlittableComponent, SpatialOSNonBlittableComponent.Update>
+        {
+            public SpatialOSNonBlittableComponentReader(Entity entity,
+                EntityManager entityManager,
+                ILogDispatcher logDispatcher)
+                : base(entity, entityManager, logDispatcher)
+            {
+            }
+
+            private readonly List<Action<string>> stringFieldDelegates = new List<Action<string>>();
+
+            [UsedImplicitly]
+            public event Action<string> StringFieldUpdated
+            {
+                add => stringFieldDelegates.Add(value);
+                remove => stringFieldDelegates.Remove(value);
+            }
+
+            private readonly List<Action<List<int>>> listFieldDelegates = new List<Action<List<int>>>();
+
+            [UsedImplicitly]
+            public event Action<List<int>> ListFieldUpdates
+            {
+                add => listFieldDelegates.Add(value);
+                remove => listFieldDelegates.Remove(value);
+            }
+
+            protected override void HandleFieldUpdates(SpatialOSNonBlittableComponent.Update update)
+            {
+                DispatchWithErrorHandling(update.StringField, stringFieldDelegates);
```

Comment:
We are technically just testing the correctness of the mock code but not whether the actual generated code works correctly, are we? Wouldn't it make more sense to test with actual generated code?